### ISSUE TITLE
Message info: detecting host provider for... is printed out although --quiet flag was set

### ISSUE THREAD

**Issue by echalone** [git-for-windows/git#3511]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

```
$ git --version --build-options

git version 2.33.1.windows.1
cpu: x86_64
built from commit: 05d80adb775077d673fab685009ede4a1003ed92
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
feature: fsmonitor--daemon
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.19043.1320]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"
> type "C:\Program Files (x86)\Git\etc\install-options.txt"
> type "%USERPROFILE%\AppData\Local\Programs\Git\etc\install-options.txt"
$ cat /etc/install-options.txt

Editor Option: Notepad++
Custom Editor Path:
Default Branch Option:
Path Option: Cmd
SSH Option: OpenSSH
Tortoise Option: false
CURL Option: OpenSSL
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Git Pull Behavior Option: Merge
Use Credential Manager: Core
Performance Tweaks FSCache: Enabled
Enable Symlinks: Disabled
Enable Pseudo Console Support: Disabled
Enable FSMonitor: Disabled
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

I was able to reproduce the issue from version 2.33.1 for windows onwards, but not with version 2.33.0.2 or the linux version.

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

Powershell and Bash

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
git clone --quiet "https://<PathToPrivateAzureDevOps2019Repo>" LocalGitRepoName
```
 - What did you expect to occur after running these commands?

No output while cloning repository (since --quiet flag is present).
Since the info line is printed out to stderr it breaks our scripts. A redirection from stderr to stdout is no solution since Powershell in Azure DevOps build agents will then throw an error and stop the script immediatly.

 - What actually happened instead?

Output of the following line to stderr:
```
info: detecting host provider for 'https://<AzureDevOps2019ServerName>/'...
```
 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

Sadly it's a private repository and I can't find a public one with which this message is printed out.
I have no problem with the message in itself, but it shouldn't be printed out if the --quiet flag is set.

**Comment by dscho** [comment]
I believe that this is fixed by microsoft/Git-Credential-Manager-Core@4472ea05595f162627d08a032b53c804a616f5e8, i.e. this report is a duplicate of https://github.com/microsoft/Git-Credential-Manager-Core/issues/492 (you had no way of knowing that this message wasn't produced by Git itself, of course).

There has not been any [Git Credential Manager release](https://github.com/microsoft/Git-Credential-Manager-Core/releases) since that fix, so you have two options: either build it yourself from the current `main` branch, or wait until a new release is available. (I am hopeful that we'll get a new release before Git for Windows v2.34.0 is due, there have been [enough changes](https://github.com/microsoft/Git-Credential-Manager-Core/compare/v2.0.567...main), I think, to merit one. @mjcheetham what's your opinion about a new Git Credential Manager Core version?)

**Comment by mjcheetham** [comment]
> I think, to merit one. @mjcheetham what's your opinion about a new Git Credential Manager Core version?)

Yes, I was planning on making one for 2.34!

**Comment by echalone** [comment]
I found yet another temporary fix (for anyone else who needs it): Providing the credential manager with a fixed provider in the configuration so it won't try to autodetect it. In this example <devops.samplecompany.com> would need to be replaced with your own domain of course.
´´´
git config --global credential.<devops.samplecompany.com>.provider generic
´´´

Thanks for your help, looking forward to the fix in the credential manager ;)

**Comment by daniol** [comment]
The milestone of this issue should be updated to "next release", as it was not included in v2.34.0
